Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -148,6 +149,21 @@ final class ConfigGenerator: ConfigGenerating {
         pbxproj.add(object: variantBuildConfiguration)
         configurationList.buildConfigurations.append(variantBuildConfiguration)
     }
+    
+    private func update(buildSettings settings: inout [String: SettingValue], byApplying manifest: Target) {
```

Comment:
Not really happy with having this function here. Any thoughts where we could put this method to be in more appropriate place? 🤔 Updating the target settings in `generateTargetSettingsFor` is quite distributed and not that clear.

Reply:
There is an `updateTargetDerived(buildSettings...)` method which applies build settings from the `Target` properties (e.g. `bundleId`, `infoPlist` etc...) - does `deploymentTarget` fit in with those?

I guess the one downside there is, it isn't in the order that you'd like (i.e. it's applied last and doesn't allow users to override it)

@pepibumur - would we have any issues moving derived settings up? (such that it's applied before user defined settings)

Reply:
> @pepibumur - would we have any issues moving derived settings up? (such that it's applied before user defined settings)

My only concern is that it's a breaking change. As a user I'd expect the derived settings to be overridable through build settings, so I think it makes sense to move it. We have to make sure  that we mark the change as potentially breaking in the CHANGELOG.

Reply:
```
// generateTargetSettingsFor(..) method
...
var settings = try defaultSettingsProvider.targetSettings(target: target,
                                                                  buildConfiguration: buildConfiguration)
updateTargetDerived(buildSettings: &settings,
                            target: target,
                            graph: graph,
                            sourceRootPath: sourceRootPath) // 👈 we will update deployment target in this method
settingsHelper.extend(buildSettings: &settings, with: target.settings?.base ?? [:])
...
```
does that make sense? ☝️ 

Could you please provide a bit more context what are `the derived settings`? 🙏

Reply:
Derived settings are the settings that are derived from the manifest. For example the deployment target, or the bundle identifier are two examples of attributes that can be set to the Target, and from which we need to *derive* the build settings that need to be generated in the resulting Xcode target.

Reply:
I implemented the requested changes 😉